src/gui/inputdevices/PenInputHandler.cpp

```diff
@@ -29,6 +29,40 @@ PenInputHandler::~PenInputHandler()
 	XOJ_RELEASE_TYPE(PenInputHandler);
 }
 
+/**
+ * Get input data relative to current input page
+ */
+PositionInputData PenInputHandler::getInputDataRelativeToCurrentPage(XojPageView* page, InputEvent* event)
+{
+    XOJ_CHECK_TYPE(PenInputHandler);
+
+    GtkXournal* xournal = inputContext->getXournal();
+
+    gdouble eventX = event->relativeX;
+    gdouble eventY = event->relativeY;
+
+    gint scale = gtk_widget_get_scale_factor(&xournal->widget);
+
+    //take scroll offset into account
+    this->inputContext->getScrollHandling()->translate(eventX, eventY);
+
+    PositionInputData pos = {};
+    pos.x = eventX - page->getX() - xournal->x;
+    pos.y = eventY - page->getY() - xournal->y;
```

Comment:
Hm, multiplying this with the scale doesn't seem to fix it either... :thinking: 

It seems to me like the relative movement is fine, but the starting point is wrong

Reply:
Just output these values to the command line and check which are wrong. Most likely you need to only multiply `eventX` and `eventY` and not the whole statement.

Reply:
Did that, but multiplying either of them with `scale` will result in the line getting longer than what I actually drew with my pen. I think the relative offset (which exentX and eventY are) are fine, but the starting point is wrong.

Reply:
Output all values of these lines to the command line and check if anything seems wrong. Maybe it is another value that is actually not set correctly for some reason.

Reply:
It could be `xournal->x` and `xournal->y`

Reply:
With no way to test I'm more or less gessing...

Reply:
So I've been tinkering about a bit, I can't really seem to find a pattern in how it differs:

```
Correct output: 
eventX:463.219 pageX17 XournalX0
eventY:314.023 pageX10 XournalX0

Output with pen (2x scale):
eventX:122.215 pageX17 XournalX0
eventY:76.625 pageX10 XournalX0
```

Done with ` std::cout << "eventX:" << eventX << " pageX" << page->getX() << " XournalX" << xournal->x << "\n";`

Reply:
You could try saving the initial starting point and adding that with some scale on top of all further calculations. It is somewhat ugly but at least it would add the constant offset you seem to require.